src/plugins/pcr_primer_design_for_dna_assembly/src/utils/PCRPrimerDesignTaskReportUtils.h

```diff
@@ -0,0 +1,106 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#ifndef _U2_PCR_PRIMER_DESIGN_TASK_REPORT_UTILS_H_
+#define _U2_PCR_PRIMER_DESIGN_TASK_REPORT_UTILS_H_
+
+#include <QStringList>
+
+namespace U2 {
+
+class PCRPrimerDesignForDNAAssemblyTask;
+struct PCRPrimerDesignForDNAAssemblyTaskSettings;
+
+struct PCRPrimerDesignTaskReportUtils {
+    // Saves all unwanted connection reports of both user primers.
+    struct UserPrimersReports;
+
+    // Returns PCRPrimerDesignForDNAAssemblyTask report. fileSequence must not be empty.
+    static QString generateReport(const PCRPrimerDesignForDNAAssemblyTask &task, const QByteArray &fileSequence,
+                                  const QByteArray &revComplSeq, const UserPrimersReports &reports);
+
+private:
+    /**
+     * To build a user primer unwanted connection table. Includes a header row, a forward user primer row, and a reverse
+     * user primer row. Number of columns varies.
+     * No validation of columns number when adding rows.
+     */
+    class UserPrimersTable;
+
+    /**
+     * Checks for unwanted user primer connections.
+     * If there are no connections, an empty string is returned.
+     * If so, information about them is returned:
+     *   selfdimer table and list of selfdimer reports,
+     *   heterodimer table and list of heterodimer reports.
+     */
+    static QString userPrimersUnwantedConnectionsInfo(const PCRPrimerDesignForDNAAssemblyTaskSettings &settings,
+                                                      const UserPrimersReports &reports);
+};
+
+struct PCRPrimerDesignTaskReportUtils::UserPrimersReports {
+    /**
+     * Stores reports of unwanted connections of a single user primer. Includes selfdimer report and heterodimer
+     * reports with sequence from file, reverse complementary sequence from file, and other sequences in PCR reaction.
+     */
+    struct PrimerReports {
+        QString selfdimer,
+            fileSeq,
+            fileRevComplSeq;
+        QStringList other;
+    };
+
+    PrimerReports forward,  // Reports of unwanted connections of forward user primer.
+        reverse;  // Reports of unwanted connections of reverse user primer.
+    QString heterodimer;  // User primers heterodimer report.
+
+    // Are there selfdimers?
+    bool hasSelfdimers() const;
+    // Are there heterodimers?
+    bool hasHeterodimers() const;
+    // Are there unwanted user primer connections?
+    bool hasUnwantedConnections() const;
+};
+
+class PCRPrimerDesignTaskReportUtils::UserPrimersTable {
+    const QString name;  // Name of table.
```

Comment:
nit: use members/private down in the class declaration

Reply:
@DmitriiSukhomlinov, @yalgaer, please see changes.